frontend/ui/widget/keyvaluepage.lua

```diff
@@ -543,6 +547,100 @@ function KeyValuePage:_populateItems()
     self.return_button:resetLayout()
     self.main_content:clear()
     local idx_offset = (self.show_page - 1) * self.items_per_page
+
+    -- for flexible middle ratio calculation
+    -- in sync with KeyValueItem actual computation
+    local frame_padding = KeyValueItem.frame_padding
+    local frame_internal_width = self.item_width - frame_padding * 2
+    local middle_padding = KeyValueItem.middle_padding
+    local available_width = frame_internal_width - middle_padding
```

Comment:
In this spirit, It would be a method that returns four values. Is that quite ok?
```
KVI.saidMethod(ratio)
return frame_internal_width, available_width, key_w, value_w
end
```

Reply:
On second thought, we can return a cached table and update it only when the input parameter has changed.

Reply:
Or let it as it is ? :)
(We don't have such things I think, or only when it is computational intensif, like rendering some text to fit in some height to get the font size that fit. A method to return 4 nearly constants makes it less readable, a cache table is not worth it.)